src/transformers/models/gpt_neox/modeling_flax_gpt_neox.py

```diff
@@ -0,0 +1,722 @@
+                tuple(batch_dims) + (1, num_updated_cache_vectors, max_length),
+            )
+            attention_mask = combine_masks(pad_mask, attention_mask)
+        return key, value, attention_mask
+
+    def _split_heads(self, hidden_states):
+        return hidden_states.reshape(hidden_states.shape[:2] + (self.num_attention_heads, self.head_size))
+
+    def _merge_heads(self, hidden_states):
+        return hidden_states.reshape(hidden_states.shape[:2] + (self.hidden_size,))
+
+    def __call__(
+        self,
+        hidden_states,
+        attention_mask,
+        position_ids,
+        deterministic: bool = True,
+        init_cache: bool = False,
+        output_attentions: bool = False,
+    ):  
+        # Compute QKV
+        # Attention heads [batch, seq_len, hidden_size]
+        #   --> [batch, seq_len, (num_heads * 3 * head_size)]
+        qkv = self.query_key_value(hidden_states)
+        batch, seq_len, _ = qkv.shape
+        # [batch, seq_len, (num_heads * 3 * head_size)]
+        #   --> [batch, seq_len, num_heads, 3, head_size]
+        qkv = qkv.reshape([batch, seq_len,self.num_attention_heads,3,self.head_size])
+        # [batch, seq_len, num_heads, 3, head_size]
+        #   --> [3,batch, seq_len, num_heads, head_size]
+        qkv = jnp.moveaxis(qkv, source=-2, destination=0)
+        # [3, batch, seq_len, num_heads, head_size]
+        #   --> [3,batch, num_heads, seq_len, head_size]
+        qkv = jnp.swapaxes(qkv, 3,2)
+        # [3,batch, num_heads, seq_len, head_size]
+        #   --> 3 [batch, num_heads, seq_len, head_size]
+        query, key, value = qkv
+
+
+        # Compute rotary embeddings on rotary_ndims
+        query_rot = query[..., : self.rotary_ndims]
+        query_pass = query[..., self.rotary_ndims :]
+        key_rot = key[..., : self.rotary_ndims]
+        key_pass = key[..., self.rotary_ndims :]
+
+        cos, sin = self.rotary_emb(value, seq_len=seq_len)
+        query, key = apply_rotary_pos_embNP(query_rot, key_rot, cos, sin, position_ids)
+        query = jnp.concatenate((query, query_pass), axis=-1)
+        key = jnp.concatenate((key, key_pass), axis=-1)
+
+
+        query_length, key_length = seq_len, seq_len
+
+        if self.has_variable("cache", "cached_key"):
+            mask_shift = self.variables["cache"]["cache_index"]
+            max_decoder_length = self.variables["cache"]["cached_key"].shape[1]
+            causal_mask = lax.dynamic_slice(
+                self.causal_mask, (0, 0, mask_shift, 0), (1, 1, query_length, max_decoder_length)
+            )
+        else:
+            causal_mask = self.causal_mask[:, :, :query_length, :key_length]
+
+        batch_size = hidden_states.shape[0]
+        causal_mask = jnp.broadcast_to(causal_mask, (batch_size,) + causal_mask.shape[1:])
+
+        attention_mask = jnp.broadcast_to(jnp.expand_dims(attention_mask, axis=(-3, -2)), causal_mask.shape)
+        attention_mask = combine_masks(attention_mask, causal_mask)
```

Comment:
But when I call generate() method, I get an error in this part. because of the swap

Reply:
Let's swap them back then after inserting the rotary embeddings? Or update the rotary embeddings to apply them to the correct axes :) Feel free to commit directly to this PR branch @HeegyuKim! Or open a new PR if you feel more comfortable doing so

Reply:
@sanchit-gandhi I have been almost resolved this problem and I'm testing implementation. So I can share it probably in 1-2weeks